inc/components/class-component.php

```diff
@@ -190,6 +190,11 @@ private function process_args( array $args ): self {
 			]
 		);
 
+		if ( ! empty( $args['config']['theme'] ) && 'irving/site-theme' !== $this->get_name() ) {
```

Comment:
I think this is doing the opposite of what we want. If I'm reading this correctly, it's taking any `theme` set inside the `config` and moving it to a top level property, rather than taking any top-level `theme` property and moving it to `config`.

The latter, is what we want so that the theme property will be accessible inside the `config_callback` and `children_callback` functions as part of the `config` object.

Reply:
The precedence could go either way, but I took it to mean enabling the ability to pass a `theme` key in the config object (which would also expose it in the callbacks). I take your point that there's a lot of existing templates that use the top-level property, so maybe the order of precedence should run the other way?